### PR TITLE
design system component library release v23.0.0

### PR DESCRIPTION
**Pull request by micahchiang**
## Description
Bumps the version of the component library to 23.0.0 in package.json. This will also roll out all the updates from v22.0.0 of the component library. 

Release notes for v23 can be found here - https://github.com/department-of-veterans-affairs/component-library/releases/tag/v23.0.0

Release notes for v22 can be found here - https://github.com/department-of-veterans-affairs/component-library/releases/tag/v22.0.0

### CODE REVIEW

**Review by rsmithadhoc — APPROVED** [review]

**Review by it-harrison — APPROVED** [review]

**Review by Asterix22 — APPROVED** [review]

**Comment by micahchiang** [comment]
@department-of-veterans-affairs/vsa-public-websites-frontend and @department-of-veterans-affairs/va-cto-health-products - Could I please get approval on this PR? 

This pull request bumps the version of the component library to v23, but I needed to fix some e2e tests that broke due to some minor changes we introduced, which is why you've been tagged as codeowners. Thank you!

**Review by harshilmoradia — APPROVED** [review]

**Review by chriskim2311 — APPROVED** [review]

**Review by mdewey — APPROVED** [review]

**Pull request merged**
